bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/CheckpointSource.java

```diff
@@ -89,7 +89,9 @@ public String toString() {
      * @param compact
      *          Flag to compact old checkpoints.
      */
-    void checkpointComplete(Checkpoint checkpoint, boolean compact) throws IOException;
+    void checkpointComplete(Checkpoint checkpoint,
+                            boolean compact,
+                            LedgerDirsManager ledgerDirsManager) throws IOException;
```

Comment:
we should add javadocs here and explain why we have this `ledgerDirsManager` and when it may be null